[ews-build] Upload layout-test logs to S3

briannafan opened the pull request:
#### 8716a6689a460ab2562117f2d75587cb8a01ad49
<pre>
[ews-build] Upload layout-test logs to S3
<a href="https://bugs.webkit.org/show_bug.cgi?id=270651">https://bugs.webkit.org/show_bug.cgi?id=270651</a>
<a href="https://rdar.apple.com/124223781">rdar://124223781</a>

Reviewed by Aakash Jain.

Adds log filtering script and upload steps.

* Tools/CISupport/ews-build/steps.py:
(RunWebKitTests):
(RunWebKitTests.__init__):
(RunWebKitTests.start):
(RunWebKitTests.evaluateCommand):
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: <a href="https://commits.webkit.org/279663@main">https://commits.webkit.org/279663@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2bb9eada55b09ef86ed23d559bba15a93c53a608

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/53622 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/32980 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/6131 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/56902 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/4348 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/55924 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/40498 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/4159 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/43449 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2840 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/55719 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/31235 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/46364 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/24584 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/28060 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/3681 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/2503 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/49789 "Passed tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/3862 "Found 1 new test failure: media/video-unmuted-after-play-holds-sleep-assertion.html (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/58497 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/28784 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/3878 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/50858 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/53625 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/29984 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/46520 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/50199 "Found 2 new API test failures: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/memory-pressure, /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/languages (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/30915 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8016 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/29761 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -3930,7 +3930,7 @@ class RunWebKitTests(shell.Test, AddToLogMixin):
 
     def __init__(self, **kwargs):
-        super().__init__(logEnviron=False, **kwargs)
+        super().__init__(logEnviron=False, timeout=19800, **kwargs)
```
better to specify the timeout explicitly as 5.5*60*60

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -4123,12 +4137,10 @@ def evaluateCommand(self, cmd):
-            self.build.addStepsAfterCurrentStep([ArchiveTestResults(),
-                                                UploadTestResults(),
-                                                ExtractTestResults()])
+            steps_to_add += [ArchiveTestResults(), UploadTestResults(), ExtractTestResults()]
```
please verify it this works as self.finished() is called before self.build.addStepsAfterCurrentStep()

briannafan replied:
Yes it works! I believe `self.finished` just sets the final status of the step.

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -3984,6 +3984,7 @@ def start(self, BufferLogObserverClass=logobserver.BufferLogObserver):
         self.log_observer_json = logobserver.BufferLogObserver()
         self.addLogObserver('json', self.log_observer_json)
         self.setLayoutTestCommand()
+        self.command = self.shell_command(' '.join(self.command) + ' 2>&1 | python3 Tools/Scripts/filter-test-logs layout')
```
Curious, can we drop python3 from this command and invoke the filter-test-logs directly?

aj062 reviewed (commented):
Mostly looks good, please make sure it's tested on uat instance.

aj062 reviewed (approved):
r+ assuming it's well tested

briannafan committed:
[ews-build] Upload layout-test logs to S3
https://bugs.webkit.org/show_bug.cgi?id=270651
rdar://124223781

Reviewed by Aakash Jain.

Adds log filtering script and upload steps.

* Tools/CISupport/ews-build/steps.py:
(RunWebKitTests):
(RunWebKitTests.__init__):
(RunWebKitTests.start):
(RunWebKitTests.evaluateCommand):
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: https://commits.webkit.org/279663@main
